### PR TITLE
ci: build changelog automatically

### PR DESCRIPTION
**Pull request by roobre**
This PR (attempts to) implement automated changelog generation for releases.

Goals:

- [x] Autogenerate changelog based on commits or, preferrably, merged PRs.
- [x] Use changelog as release body.
- [x] Generate a CHANGELOG file and put it in the archives.

### Feedback

The generated changelog for the singe release can be found [here](https://github.com/newrelic/nri-docker/releases/tag/v9.9.9)

The generated full changelog can be viewed with the following command:

```bash
curl -L https://github.com/newrelic/nri-docker/releases/download/v9.9.9/nri-docker_9.9.9-1_amd64.deb | tar -xz --to-stdout data.tar.gz | tar -xz --to-stdout ./usr/share/doc/nri-docker/CHANGELOG.md
```

Or in this gist: https://gist.github.com/roobre/ae85275179856928c4807e5033865eaa

Appending release notes can be done by opening a "meta-issue", which is a bit cumbersome but still doable. The action has support for noticing features, bugfixes, and breaking changes depending on issue labels. The full capabilities of the generator can be seen [here](https://github.com/github-changelog-generator/github-changelog-generator)

### CODE REVIEW

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -88,11 +88,24 @@ jobs:
...
           username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
           password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Create CHANGELOG.md
+        uses: heinrichreimer/github-changelog-generator-action@v2.2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          issues: true
+          issuesWoLabels: true
+          pullRequests: true
+          prWoLabels: true
```
Comment (roobre) [inline .github/workflows/prerelease.yml:105]: These four shouldn't need to be specified. A bug has been reported upstream: https://github.com/heinrichreimer/action-github-changelog-generator/issues/23

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -88,11 +88,24 @@ jobs:
...
       - name: Login to DockerHub
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
           password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+      - name: Create CHANGELOG.md
+        uses: heinrichreimer/github-changelog-generator-action@v2.2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          issues: true
```
Comment (gsanchezgavier) [inline .github/workflows/prerelease.yml:102]: i would not add issues. We will end up populating the changelog with things that are not related to code changes like [this one ](https://github.com/newrelic/nri-docker/issues/47)

**Comment by gsanchezgavier** [comment]
Excellent tool , this will automate the changelog and give it richer info. I just have a concern about the issues that are added in the changelog.

File: .github/workflows/prerelease.yml (hunk 2 above)
Comment (roobre) [reply .github/workflows/prerelease.yml:102]: I can try to add issues but only those with labels (`bug`, `enhancement`, etc.) and see how it comes.

**Comment by arvdias** [comment]
Looks good. Is the PRs section always named "Merged pull requests" ?  Ideally we would have sections named "Added", "Changed" and "Fixed" but i guess we can't control this very well. I think I had tried this tool before and didn't see how we could do that, although I didn't spend much time with it.

**Comment by roobre** [comment]
Due to the increased complexity and upstream bugs we're dropping this from the epic for now. Closing.
